### PR TITLE
ingestion-post-and-get-api-changes

### PR DESCRIPTION
**Pull request by gopal-mahajan**
- **API Server Changes**

1.  Changes done in registerAdapter method and publishAllAdapterForUser method

- **ManagmentApiImpl Changes**

1. In registerAdapter method flow to create exchange has been refactored to cache call for given entities followed by insertion in postgres which is further followed by creation of exchange in dataBroker.

2. In publishAllAdapterForUser, call to dataBroker has been discontinued. Now data is being fetched from a call to postgres.

**Note** : ManagmentApiImplTest is commented as changes are yet to be made in that file.

### CODE REVIEW

File: src/main/resources/db/migration/V4_5__Create_ingestion_table.sql [hunk 1]
```diff
@@ -0,0 +1,10 @@
+CREATE TABLE IF NOT EXISTS ingestion
```
Comment (kailash) [inline src/main/resources/db/migration/V4_5__Create_ingestion_table.sql:1]: 1. rename DB table name to `adaptors_details`.
2. add `created_at` and `modified_at` columns
3. insert `created_at` and `modified_at` columns through trigger on insert/update

File: src/main/java/iudx/resource/server/apiserver/management/ManagementApiImpl.java [hunk 2]
```diff
@@ -302,7 +352,7 @@ public Future<JsonObject> registerAdapter(JsonObject json, DataBrokerService dat
    */
   @Override
   public Future<JsonObject> deleteAdapter(String adapterId, String userId,
```
Comment (kailash) [inline src/main/java/iudx/resource/server/apiserver/management/ManagementApiImpl.java:354]: Once Adapter is deleted from RMQ, is should also get deleted from the database.

**Review by kailash — CHANGES_REQUESTED** [review]

File: src/main/java/iudx/resource/server/apiserver/management/ManagementApiImpl.java [hunk 3]
```diff
@@ -448,19 +498,21 @@ public Future<JsonObject> publishDataFromAdapter(JsonObject json, DataBrokerServ
   }
 
   @Override
-  public Future<JsonObject> publishAllAdapterForUser(JsonObject request, DataBrokerService databroker) {
+  public Future<JsonObject> publishAllAdapterForUser(JsonObject request, PostgresService postgresService) {
```
Comment (kailash) [inline src/main/java/iudx/resource/server/apiserver/management/ManagementApiImpl.java:501]: The method name `getAllAdapterDetailsForUser` makes more sense than `publishAllAdapterForUser`. Change method name to `getAllAdapterDetailsForUser`

**Comment by gopal-mahajan** [comment]
image

Screenshot of code coverage.

**Comment by kailash** [comment]
retest this please

**Comment by kailash** [comment]
retest this please

**Comment by kailash** [comment]
retest this please

**Pull request merged**